### PR TITLE
[None][bug] fix SM90 full-mask skip-softmax dispatch

### PR DESCRIPTION
**Pull request by bobboli**
## Summary
- fix an SM90 FMHA V2 dispatch hash collision between `attention_mask_type` and `enable_skip_softmax`
- restore the correct full-mask skip-softmax kernel selection instead of relying on a Python-side guard
- add an SM90 regression test that exercises full-mask prefill with skip-softmax and verifies threshold forwarding

## Root cause
`ContextAttentionMaskType::CUSTOM_MASK` needs 3 bits in the FMHA V2 hash. The previous packing overlapped the mask-type field with the skip-softmax bit, so `PADDING + skip-softmax` collided with `CUSTOM_MASK + no skip-softmax` and dispatched the wrong kernel on SM90.

## Validation
- `pytest -q tests/unittest/_torch/attention/test_skip_softmax_sm90.py`
- `python work/20260416/repro_skip_softmax_sm90.py --mask full --skip-softmax --disable-sm90-full-mask-guard`
- `compute-sanitizer --tool memcheck --target-processes all python work/20260416/repro_skip_softmax_sm90.py --mask full --skip-softmax --disable-sm90-full-mask-guard`

## Summary by CodeRabbit

* **Refactor**
  * Improved internal code maintainability in attention kernel implementation through named constants and validation checks.

* **Tests**
  * Added SM90-specific unit tests to validate skip-softmax attention configuration and behavior.

### CODE REVIEW

**Comment by karljang** [comment]
I confirmed that this fix works on H200 devices.
Thank you!

**Review by karljang — APPROVED** [review]
